extensions/wikia/IndexingPipeline/tests/PipelineEventProducerTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+class PipelineEventProducerTest extends WikiaBaseTest {
+	private $pipelineEventProducer;
+
+	protected function setUp() {
+		$this->setupFile = dirname( __FILE__ ) . '/../PipelineEventProducer.setup.php';
+		$this->pipelineEventProducer = new PipelineEventProducer();
+		parent::setUp();
+	}
+
+	/** @test
+	 * @param $inputData
+	 * @param $expectedOutput
+	 * @param $description
+	 * @dataProvider prepareRouteTestDataProvider
+	 */
+	public function prepareRouteTest($inputData, $expectedOutput, $description) {
+		$actualOutput = $this->pipelineEventProducer->prepareRoute( $inputData['action'], $inputData['ns'], $inputData['flags'], $inputData['data'] );
+
+		$this->assertEquals( $expectedOutput, $actualOutput, $description );
+	}
+
+	public function prepareRouteTestDataProvider() {
+		global $wgCanonicalNamespaceNames;
+		return [
+			[
+				'data' => [
+					'action' => PipelineEventProducer::ACTION_CREATE,
+					'ns' => strtolower( $wgCanonicalNamespaceNames[ NS_TEMPLATE ] ),
+					'flags' => [],
```

Comment:
you may want to test also the calls that do not have the necessary params provided (nulls)